docs/dev/table/streaming.md

```diff
@@ -39,17 +39,293 @@ Dynamic table
 Time Attributes
 ---------------
 
-### Event-time
+Flink supports different notions of *time* in streaming programs.
 
-* DataStream: Timestamps & WMs required, `.rowtime` (replace attribute or extend schema)
-* TableSource: Timestamps & WMs & DefinedRowtimeAttribute
+- *Processing time* refers to the system time of the machine (also known as "wall-clock time") that is executing the respective operation.
+- *Event time* is the time that each individual event occurred on its producing device.
+- *Ingestion time* is the time that events enter Flink, internally, it is treated similar to event time.
```

Comment:
*Ingestion time* is the time that events enter Flink; internally, it is treated similarly to event time.